packages/syntax-core/src/IconButton/IconButton.tsx

```diff
@@ -1,9 +1,12 @@
+"use client";
+
```

Comment:
@benjroy we should start adding `"use client"` to all of our components which are interactive.
See https://github.com/mui/material-ui/pull/37656 for more info

`"use client"` means that the component can still be rendered on the server but they will also be included in our JS bundles so that they can re-render on the client.

[our JS bundles](https://www.joshwcomeau.com/react/server-components/#specifying-client-components-5)

Reply:
Is this the only file in the repo that needs this, or is this absolutely necessary to have on this file in this PR in order for these changes to work?

If it is not, maybe a separate PR would be more appropriate so that we don't miss anything in the context of this statement:

> we should start adding "use client" to all of our components which are interactive.